conbench/entities/benchmark_result.py

```diff
@@ -1283,13 +1288,47 @@ class _BenchmarkResultCreateSchema(marshmallow.Schema):
     context = marshmallow.fields.Dict(  # type: ignore[assignment]
         required=True,
         metadata={
-            "description": "Information about the context the benchmark was run in (e.g. compiler flags, benchmark langauge) that are reasonably expected to have an impact on benchmark performance. This information is expected to be the same across a number of benchmarks. (free-form JSON)"
+            "description": conbench.util.dedent_rejoin(
+                """
+                Required. Must be a JSON object (empty dictionary is allowed).
+
+                Relevant benchmark context (other than hardware/platform
+                details and benchmark case parameters).
+
+                Conbench requires this object to remain constant when doing
+                automated timeseries analysis (this breaks history).
+
+                Use this to store for example compiler flags or a runtime
+                version that you expect to have significant impact on
+                measurement results.
```

Comment:
Oh yes, great point.

Reply:
Yeah, users frequently screw this up when assembling result JSON, so it should definitely be documented. Previously I've advocated for validating it too so users get better errors, but I think we made it optional in certain cases now?

Reply:
>a generic interface for ingesting measurement data

Here's my roundabout logic for caring about code language. Your quote above is definitely a core value of Conbench. But I think we agree that another core value is Conbench's ability to aid in the analysis of changes in that data, and further, connect those changes to code changes in a git repository. Making the assumption that every Conbench user deeply cares about this last point is what enables me to think about tailoring our interfaces to that use case (instead of, say, measuring water level 😉). Therefore I think a great "tailorization" is to embrace the fundamentals of coding in a git repository, like code language.

Reply:
💯 That's why I get uneasy when we accept results with missing git metadata—because our analysis indexes by commit, these results largely get ignored unless users analyze them themselves.

Reply:
I think we have a misunderstanding here :) I come from a product description / marketing / messaging perspective.

> I think a great "tailorization" is to embrace the fundamentals of coding in a git repository, like code language.

I agree.

When I wrote above "I find it to be not the strongest selling point" I literally meant that it is not our strongest selling point. Yes it is a selling point, just not particularly strong relative to others.

A prospect reads

> ability to manage different benchmarking languages

> Conbench allows you to write benchmarks in any language

> Language-independent Continuous Benchmarking

That raises pretty big expectations with respect to this language-independent nature; and it raises the question "what do they mean by that, what's the secret sauce here, how much work was it to achieve this?"

As of today I think this is what we mean:

- we provide a generic interface (for other good reasons than just language agnosticism)
- when having `benchmark_language` in `context` then the UI ...

That's about it. Right? What do I miss?

> Conbench's ability to aid in the analysis of changes in that data, and further, connect those changes to code changes in a git repository

That type of stuff is what I was referring to with "actual value " when writing above

> distracting from actual value 

That 'actual value ' was of course non-ideal wording because it implies that the other thing has little value. I should have said "from our stronger selling points" :).

Reply:
> I get uneasy when we accept results with missing git metadata—because our analysis indexes by commit, these results largely get ignored unless users analyze them themselves.

Haha. Well, one can add some structure to this and then one can be almost worry-free?

Can be reconciled:

- low getting-started/testing barrier (not requiring all the fields)
- documenting what's needed for achieving what (we've made good progress here in the recent months)

>  these results largely get ignored unless users analyze them themselves.

that's pretty much what we document as of today

Reply:
These are great thoughts as well! Agreed that our marketing pitch should be better-tuned.

Reply:
> Can be reconciled:
> 
> * low getting-started/testing barrier (not requiring all the fields)
> * documenting what's needed for achieving what (we've made good progress here in the recent months)

Agreed! The hidden cost here of starting with incomplete metadata is that adding it later will often break histories. (Not applicable to git info, without which there can be no history, but certainly `context`, `tags`, hardware, etc.) Lowering this impact would be good. 

Idle thought: I wonder if we should allow the modification of metadata for all results in a timeseries fingerprint (e.g. add/remove/modify a case tag) so it merges with another? I can think a lot of existing use-cases, e.g. adding batch name to Archery microbenchmark benchmark name, populating CPU count in Arrow R benchmarks 😭, removing vestigial tags when arrowbench benchmarks are run via adapters, etc. etc.